### PR TITLE
Loc templates

### LINKED ISSUE #5249: [Loc]With net6 preview 7 installed on locale OS, some template names are unlocalized after runing "dotnet new --list" in CLI .
1. Install net 6 preview 7 SDK[6.0.100-preview.7.21364.30(runtime:6.0.0-preview.7.21360.10)] from https://github.com/dotnet/installer
2. Install it on locale OS
3. Run “dotnet new --list” in CLI
4. Check if  output contents are localized well.
Actual Result:
Some template names are not localized.

e.g 
SC:
image

DE:
image

Note:
It has been localized well in VS 17.0 preview 3
e,g
SC:
image

### PR DESCRIPTION
**Pull request by RussKie**
Resolves #5249

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5429)

### CODE REVIEW

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.cs.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (vlada-shubina) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.cs.json:1]: Only english json should be generated - the rest will be added by OneLoc build

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-VisualBasic/.template.config/template.json [hunk 2]
```diff
@@ -142,6 +143,7 @@
         "continueOnError": true
       },
       {
+        "id": "editor",
```
Comment (vlada-shubina) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-VisualBasic/.template.config/template.json:146]: nth: I would use 'open-file' to be consistent with common projects templates.
https://github.com/dotnet/templating/blob/9d8b5997cab40cf2bc6ac082491be08295c5dba4/template_feed/Microsoft.DotNet.Common.ProjectTemplates.6.0/content/ClassLibrary-CSharp/.template.config/template.json#L143

**Review by vlada-shubina — COMMENTED** [review]
Consider adding display name for symbols to be localized. At the moment they are not used however they likely to be used by IDE/CLI soon.

See: https://github.com/dotnet/templating/blob/9d8b5997cab40cf2bc6ac082491be08295c5dba4/template_feed/Microsoft.DotNet.Common.ProjectTemplates.6.0/content/ClassLibrary-CSharp/.template.config/template.json#L67
as example.

**Comment by RussKie** [comment]
Thank you @vlada-shubina, updated.

**Comment by phenning** [comment]
One thing, FYI. The localizations for the net framework choices won't be used by Visual Studio, as we instead have consistent display names there across all TFMS - including third party, so we ignore the display name for that particular parameter.

**Review by dreddy-work — APPROVED** [review]

**Comment by RussKie** [comment]
Our templates are used for .NET 5.0 and .NET 6.0 apps, so we can't remove old locs. At some point we could consider separating .NET 5.0 templates, like we do for .NET Core 3.1.

**Comment by phenning** [comment]
@RussKie The targetted TFMs don't matter here. The ship vehicle of the template pack does. As long as this template pack with the template change is not being shipped in a version of the SDK which is included with Visual Studio 2019, it would be safe to remove the strings.json regardless, since Visual Studio 2019 will not try to use the template packs installed under %ProgramFiles%\dotnet\templates\6.x, since NET 6 is no longer supported in Visual Studio 2019.

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/template.json [hunk 3]
```diff
@@ -53,7 +58,8 @@
       "datatype": "text",
       "description": "Sets langVersion in the created project file",
       "defaultValue": "",
-      "replaces": "$(ProjectLanguageVersion)"
+      "replaces": "$(ProjectLanguageVersion)",
+      "displayName": "Language Version"
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/template.json:62]: @Tanya-Solyanik good catch, I intended to use sentence casing as VS uses sentence casing. I will fix common project templates shortly.

**Pull request merged**